### PR TITLE
Add graphics_labeler for labeling PRs with image changes

### PR DESCRIPTION
**Pull request by skateman**
We would like to use this for detecting image changes in the `ui-classic` repo. FYI @epwinchell

### CODE REVIEW

File: spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+describe CommitMonitorHandlers::CommitRange::GraphicsLabeler do
+  let(:branch)         { create(:pr_branch) }
+  let(:git_service)    { double("GitService", :diff => double("RuggedDiff", :new_files => new_files)) }
+
+  before do
+    stub_sidekiq_logger
+    stub_settings(:graphics_labeler => {:enabled_repos => [branch.repo.name]})
+    expect_any_instance_of(Branch).to receive(:git_service).and_return(git_service)
```
Comment (chrisarcand) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: ❌ 

`allow(branch).to receive...`

Comment (skateman) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: Probably it's not saving me, but I copied this [from here](https://github.com/ManageIQ/miq_bot/blob/master/spec/workers/commit_monitor_handlers/commit_range/gem_changes_labeler_spec.rb#L8) :D fixing

Comment (skateman) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: @chrisarcand hm, seems like it fails if i change it that way :confused:

Comment (chrisarcand) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: I figured ;) BUT NO THAT DOESN'T SAVE YOU 😜

Comment (chrisarcand) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: You need to stub https://github.com/ManageIQ/miq_bot/blob/master/app/workers/concerns/branch_worker_mixin.rb#L6-L20

To return the branch you're created here.

Comment (skateman) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: Will check it tomorrow, thanks!

Comment (Fryguy) [inline spec/workers/commit_monitor_handlers/commit_range/graphics_labeler_spec.rb]: Another option is just to test process_branch directly. psuedo-code idea.  This is part of the reason most of the workers have a separate process_branch method from the perform method (although I never got around to writing the tests this way).

```ruby
subject = CommitMonitorHandlers::CommitRange::GraphicsLabeler.new
subject.branch = branch
allow(branch).to receive #...
```

**Comment by Fryguy** [comment]
Overall LGTM :+1:  Let's work out the specs discussion, then I am good with merging this.

**Comment by Fryguy** [comment]
@chrisarcand Is #316 just a generic version of this?  Does it make sense to implement this as a rule of #316?

**Comment by chrisarcand** [comment]
Oh hah, I totally forgot that this thing probably did half of what I needed to do for #316. Whoops. Yeah I think we could probably tweak it to cover this case.

**Comment by chrisarcand** [comment]
Yeah @skateman I completely unknowingly just recreated a much more generalized version of this; I think I'll finish that instead as there are other purposes for this sort of thing that we want anyway. 💙

**Comment by skateman** [comment]
@chrisarcand no problems, less work for me ^^ :wink:

**Pull request closed without merging**